jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/pinning/ConnectionPinningManager.kt

```diff
@@ -97,12 +97,13 @@ class DefaultConnectionPinningManager(private val project: Project) :
             "${features.dropLast(1).joinToString(",") { it.featureName }} and ${features.last().featureName}"
         }
 
+        var connectionToPin = oldConnection
         if (computeOnEdt { showDialogIfNeeded(oldConnection, newConnection, featuresString) }) {
             features.forEach {
-                setPinnedConnection(it, oldConnection)
+                connectionToPin = if (it.supportsConnectionType(oldConnection)) oldConnection else newConnection
+                setPinnedConnection(it, connectionToPin)
             }
-
-            notifyInfo(message("credentials.switch.notification.title", featuresString, oldConnection.label))
+            notifyInfo(message("credentials.switch.notification.title", featuresString, connectionToPin.label))
```

Comment:
this notification message seems inconsistent with what's actually happening

Reply:
could you please elaborate?

Reply:
the `if (it.supportsConnectionType(oldConnection)) oldConnection else newConnection` condition is confusing

basically we're trying to figure out if the old or new connection is the appropriate one to pin?
if so, can it be determined once instead of in the for-loop?

Reply:
yes that is right, sure let me move that out of the for-loop

Reply:
I see the notify part is aggregated, moved it outside of the for loop